app/controllers/vm_common.rb

```diff
@@ -1241,9 +1241,10 @@ def get_node_info(treenodeid)
     else      # Get list of child VMs of this node
       options = {:model => model}
       if x_node == "root"
-        # TODO: potential to move this into a model with a scope built into it
-        options[:where_clause] =
-          ["vms.type IN (?)", ManageIQ::Providers::InfraManager::Vm.subclasses.collect(&:name) + ManageIQ::Providers::InfraManager::Template.subclasses.collect(&:name)] if x_active_tree == :vandt_tree
+        if x_active_tree == :vandt_tree
+          klass = ManageIQ::Providers::InfraManager::VmOrTemplate
+          options[:where_clause] = ["vms.type IN (?)", klass.vm_descendants.collect(&:name)]
```

Comment:
`:where_clause` gets into the session. so I get squeamish around changes like this.

I like the hash much better.

I'll make that change and ensure this tab still works for me